### ISSUE TITLE
max-len rule gives false negative with ignoreStrings set to true on certain circumstances

### ISSUE THREAD

**Issue by enrique-ramirez** [eslint/eslint#9910]
**Tell us about your environment**

* **ESLint Version:**
v4.16.0

* **Node Version:**
v8.9.4

* **npm Version:**
v5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<summary>Configuration</summary>

```
"max-len": ["error", 100, 2, {
  ignoreUrls: true,
  ignoreComments: true,
  ignoreRegExpLiterals: true,
  ignoreStrings: true,
  ignoreTemplateLiterals: true
}],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

With the above configuration, the following line is NOT throwing an error:

```js
export const myConst = someFunc(() => value => (an.object.with.someComputation() ? undefined : 'a string at the end'))
```

This does show an error, though:

```js
const _a_string_at_the_end_ = 'someString'
export const myConst = someFunc(() => value => (an.object.with.someComputation() ? undefined : _a_string_at_the_end_))
```

```bash
eslint path/to/file.js
```

**What did you expect to happen?**
It should throw an error stating that the line length is too long.

**What actually happened? Please include the actual, raw output from ESLint.**
The string at the end of the terniary operator seems to confuse eslint into believing it's a whole string, and allows the longer-than-allowed line.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. However, this is working as intended -- the `ignoreStrings` option makes the rule ignore lines that contain string literals.

**Comment by enrique-ramirez** [comment]
Ah, I see.

I think this kinda makes sense, though it was a bit surprising. Sorry for the trouble!

**Comment by kaicataldo** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
